src/Components/Activities/index.jsx

```diff
@@ -40,16 +120,70 @@ const Activities = () => {
     setShowModal(false);
   };
 
+  const closeErrorModal = () => {
+    setShowErrorModal(false);
+  };
+
+  const closeAddModal = () => {
+    setShowAddModal(false);
+  };
+
+  const closeAddErrorModal = () => {
+    setShowAddErrorModal(false);
+  };
+
+  const handleShowForm = () => {
+    setShowForm(true);
+  };
+
+  const handleEditItem = (activity, id) => {
+    setSelectedActivity(activity);
+    setShowEditForm(true);
+    setActivityId(id);
+  };
+
   return (
     <section className={styles.container}>
       <h2>Activities</h2>
-      <button className={styles.addButton}> + Add new activity</button>
+      <button className={styles.addButton} onClick={handleShowForm}>
+        + Add new activity
+      </button>
       <Modal
```

Comment:
Then, instead of having several functions like closeModal, closeErrorModal, closeAddModal, closeAddErrorModal, you can use the function handleCloseModal and pass it as parameter the function that changes the state of the modal you want to close
```suggestion
<Modal
  title="The activity has been successfully deleted!"
  show={showModal}
  closeModal={() => handleCloseModal(setShowModal)}
/>
```

Reply:
Fixed! thanks you for the advice!